### PR TITLE
Fix F# starter web template.

### LINKED ISSUE #44729: mvc F# template: conditions defined in the templates are not correct
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

`mvc` F# template: conditions defined in the templates are not correct.

I accidentally found out that conditions defined in F# `mvc` template are not correct. 
Example of incorrect conditions:
https://github.com/dotnet/aspnetcore/blob/1e8fe7e4ba93935ebc10aee6ef2a4d2b3b0c05d9/src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-FSharp/Program.fs#L13
https://github.com/dotnet/aspnetcore/blob/1e8fe7e4ba93935ebc10aee6ef2a4d2b3b0c05d9/src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-FSharp/Program.fs#L39

F# files (same as C# files) use C++ condition evaluator, which requires the condition to be in brackets. 
If the template is run in `Debug` mode, it will fall to exception.
In release mode, the condition is attempted to be evaluated anyway, and it works (likely accidentally), so it lowers the priority.

Conditions in C# files are correct.

### Expected Behavior

Use `#if (HasHttpsProfile)` syntax for conditions in F# files, similar to C# files.

### Steps To Reproduce

Run `dotnet new mvc --language F#` template in Debug mode of dotnet sdk.

### Exceptions (if any)

_No response_

### .NET Version

.NET 6, NET 7

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mitchdenny**
Fixes https://github.com/dotnet/aspnetcore/issues/44729

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-FSharp/Program.fs [hunk 1]
```diff
@@ -36,7 +36,7 @@ module Program =
 
         if not (builder.Environment.IsDevelopment()) then
             app.UseExceptionHandler("/Home/Error")
-#if HasHttpsProfile
+#if (HasHttpsProfile)
```
Comment (eerhardt) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-FSharp/Program.fs:39]: What about line 13?

Comment (eerhardt) [reply src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-FSharp/Program.fs:39]: Do we have this issue in any other F# templates?

Comment (mitchdenny) [reply src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-FSharp/Program.fs:39]: We did. Thanks for mentioning it. I've fixed them up. I've also fixed up line 13.

**Review by eerhardt — APPROVED** [review]
LGTM

**Pull request merged**